src/rate_limiter/src/persist.rs

```diff
@@ -72,10 +75,12 @@ impl Persist<'_> for RateLimiter {
     fn restore(_: Self::ConstructorArgs, state: &Self::State) -> Result<Self, Self::Error> {
         let rate_limiter = RateLimiter {
             // Safe to unwrap because TokenBucket::restore doesn't return errors.
-            ops: state
-                .ops
-                .as_ref()
-                .map(|ops| TokenBucket::restore((), ops).unwrap()),
+            ops: state.ops.as_ref().map(|ops| {
+                TokenBucket::restore((), ops).unwrap_or_else(|e| {
+                    error!("Failed to restore rate limiter: {}", e);
+                    std::process::exit(i32::from(EXIT_CODE_GENERIC_ERROR));
```

Comment:
I think the error should be propagated upwards and handled somewhere else, but I'm not very familiar with the snapshoting code.